test/test_force.js

```diff
@@ -118,63 +134,93 @@ function main(args) {
     }
 
     // Test all the platforms / app types requested
-    for (var i=0; i<chosenOperatingSystems.length; i++) {
-        var os = chosenOperatingSystems[i];
-        if (testingWithAppType) {
-            for (var j=0; j<chosenAppTypes.length; j++) {
-                var appType = chosenAppTypes[j];
-                createCompileApp(tmpDir, os, appType, null, pluginRepoUri, useSfdxRequested);
+    if (testingWithClis) {
+        for (var i=0; i<chosenClis.length; i++) {
+            var cliName = chosenClis[i];
+            var templates = templateHelper.getTemplates(SDK.forceclis[cliName]);
+            for (var j=0; j<templates.length; j++) {
+                var template = templates[j];
+                for (var k=0; k<template.platforms.length; k++) {
+                    var os = template.platforms[k];
+                    createCompileApp(tmpDir, os, template.appType, template.url, null, useSfdxRequested);
+                }
             }
         }
-        if (testingWithTemplate) {
-            // NB: chosenAppTypes[0] is getAppTypeFromTemplate(templateRepoUri)
-            createCompileApp(tmpDir, os, chosenAppTypes[0], templateRepoUri, null, useSfdxRequested);
+    }
+    else {
+        for (var i=0; i<chosenOperatingSystems.length; i++) {
+            var os = chosenOperatingSystems[i];
+            if (testingWithAppType) {
+                for (var j=0; j<chosenAppTypes.length; j++) {
+                    var appType = chosenAppTypes[j];
+                    createCompileApp(tmpDir, os, appType, null, pluginRepoUri, useSfdxRequested);
+                }
+            }
+
+            if (testingWithTemplate) {
+                createCompileApp(tmpDir, os, templateHelper.getAppTypeFromTemplate(templateRepoUri), templateRepoUri, null, useSfdxRequested);
+            }
         }
     }
 }
 
 //
 // Usage
 //
-function usage(exitCode) {
+function shortUsage(exitCode) {
```

Comment:
Short usage is printed when there is a validation error.
Full usage is printed when --usage is requested by user.